### PR TITLE
Add README for db_stress

### PR DESCRIPTION
**Pull request by cheng-chang**
This document describes the architecture and code structure of db_stress, in the hope of making it easier to add new tests.

Test Plan:
N/A

### CODE REVIEW

**Comment by siying** [comment]
Can we move it as a part of the code comment? It's hard for users to discover and to maintain this file otherwise.

File: db_stress_tool/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+then coordiates the threads to concurrently do operations on the DB and verify the DB content at certain time points.
+
+## Statistics
+Statistics (such as the number of operations) about the test are defined as `Stats` in `db_stress_stat.h`.
+They are collected during running the tests. After finishing the test, they are reported to stdout.
+
+## Fault Injection
+The stress test is also called the crash test. It crashes the testing threads/process randomly.
+There are 2 kinds of crash: blackbox and whitebox.
+Blackbox crash means `kill -9` the testing process randomly.
```
Comment (siying) [inline db_stress_tool/README.md:42]: This is outside stress test and is a part of "crash_test". Maybe should move to the commend there.

File: db_stress_tool/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Whitebox crash means crash at predefined points in certain code path.
+For example, `utilities/fault_injection_fs.h` declares a set of classes related to filesystem to simulate crash when calling certain filesystem operations.
```
Comment (siying) [inline db_stress_tool/README.md:44]: This feels as if it is a part of whitebox crash test but it is not related.

**Comment by cheng-chang** [comment]
> Can we move it as a part of the code comment? It's hard for users to discover and to maintain this file otherwise.

Are you suggesting to move this to file such as `db_stress.cc` as file level comment?

**Comment by cheng-chang** [comment]
@siying Since the readme describes the overall code structure, I think it makes sense to go to a directory level readme. If we only add docs to independent classes, people new to the module still have no idea how the different pieces work together. What do you think?

**Comment by cheng-chang** [comment]
@siying ping

**Pull request closed without merging**